textdb/textdb-web/src/main/java/edu/uci/ics/textdb/web/request/beans/OperatorBean.java

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * This class is the abstract class that defines the data members common to all operators. It is
+ * extended by individual operator beans in order to define the data members specific to each
+ * operator
+ * Created by kishorenarendran on 10/21/16.
+ */
+@JsonTypeInfo(use=JsonTypeInfo.Id.NAME, include= JsonTypeInfo.As.PROPERTY, property="operator_type", visible = true)
+@JsonSubTypes({
+        @JsonSubTypes.Type(value=DictionaryMatcherBean.class, name="DictionaryMatcher"),
+})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public abstract class OperatorBean {
+    @JsonProperty("operator_id")
+    private String operatorID;
+    @JsonProperty("operator_type")
+    private String operatorType;
+    @JsonProperty("attributes")
+    private String attributes;
+    @JsonProperty("limit")
+    private String limit;
+    @JsonProperty("offset")
+    private String offset;
+
+    public OperatorBean() {
+    }
+
+    public OperatorBean(String operatorID, String operatorType, String attributes, String limit, String offset) {
+        this.operatorID = operatorID;
+        this.operatorType = operatorType;
+        this.attributes = attributes;
+        this.limit = limit;
+        this.offset = offset;
+    }
+
+    @JsonProperty("operator_id")
+    public String getOperatorID() {
+        return operatorID;
+    }
+
+    @JsonProperty("operator_id")
+    public void setOperatorID(String operatorID) {
+        this.operatorID = operatorID;
+    }
+
+    @JsonProperty("operator_type")
+    public String getOperatorType() {
+        return operatorType;
+    }
+
+    @JsonProperty("operator_type")
+    public void setOperatorType(String operatorType) {
+        this.operatorType = operatorType;
+    }
+
+    @JsonProperty("attributes")
+    public String getAttributes() {
+        return attributes;
+    }
+
+    @JsonProperty("attributes")
+    public void setAttributes(String attributes) {
+        this.attributes = attributes;
+    }
+
+    @JsonProperty("limit")
+    public String getLimit() {
+        return limit;
+    }
+
+    @JsonProperty("limit")
+    public void setLimit(String limit) {
+        this.limit = limit;
+    }
+
+    @JsonProperty("offset")
+    public String getOffset() {
+        return offset;
+    }
+
+    @JsonProperty("offset")
+    public void setOffset(String offset) {
+        this.offset = offset;
+    }
+
+    public HashMap<String, String> getOperatorProperties() {
+        HashMap<String, String> basicOperatorProperties = new HashMap<String, String>();
+        if(this.getLimit() != null)
+            basicOperatorProperties.put(OperatorBuilderUtils.LIMIT, this.getLimit());
+        if(this.getOffset() != null)
+            basicOperatorProperties.put(OperatorBuilderUtils.OFFSET, this.getOffset());
+        basicOperatorProperties.put(OperatorBuilderUtils.ATTRIBUTE_NAMES, this.getAttributes());
+        return basicOperatorProperties;
+    }
+
+    @Override
+    public boolean equals(Object other){
```

Comment:
I'm just wondering where do we introduce the dependency of `org.apache.commons`? Does it come with dropwizard?